### PR TITLE
example of kaggle loader

### PR DESCRIPTION
**Pull request by ilyashnil**

### CODE REVIEW

File: src/unitxt/loaders.py [hunk 1]
```diff
@@ -21,6 +22,14 @@
...
     ibm_boto3_available = False
 
+try:
+    import opendatasets
+
+    opendatasets_available = True
+
+except ImportError:
+    opendatasets_available = False
+
```
Comment (elronbandel) [inline src/unitxt/loaders.py:32]:
```suggestion
```

This should be in the verify() function of the loader

File: src/unitxt/loaders.py [hunk 2]
```diff
@@ -98,6 +107,28 @@ def process(self):
         )
 
 
+# TODO write how to obtain kaggle credentials
+class LoadFromKaggle(Loader):
+    url: str
+
+    def verify(self):
+        super().verify()
+        assert opendatasets_available, "Please install opendatasets in order to use the LoadFromKaggle loader (using `pip install opendatasets`) "
```
Comment (elronbandel) [inline src/unitxt/loaders.py:116]:
```suggestion
        super().verify()
        try:
            import opendatasets
        except ImportError as e:
             raise ImportError("Please install opendatasets in order to use the LoadFromKaggle loader (using `pip install opendatasets`) ") from e
```

File: src/unitxt/loaders.py [hunk 3]
```diff
@@ -98,6 +107,28 @@ def process(self):
...
+        assert os.path.isfile(
+            "kaggle.json"
+        ), "Please obtain kaggle credentials https://christianjmills.com/posts/kaggle-obtain-api-key-tutorial/ and save them to local ./kaggle.json file"
+
+    def _download_from_kaggle(self, url, data_dir):
+        opendatasets.download(url, data_dir)
```
Comment (elronbandel) [inline src/unitxt/loaders.py:122]:
```suggestion
    def prepare(self):
        super().prepare()
        from opendatasets import download
        self.downloader = download
```

File: src/unitxt/loaders.py [hunk 4]
```diff
@@ -98,6 +107,28 @@ def process(self):
...
+
+    def process(self):
+        with TemporaryDirectory() as temp_directory:
+            self._download_from_kaggle(self.url, temp_directory)
```
Comment (elronbandel) [inline src/unitxt/loaders.py:126]:
```suggestion
            self.downloader(self.url, temp_directory)
```

**Review by elronbandel — CHANGES_REQUESTED** [review]
Overall looks great. Just refer to the minor changes I suggested.

**Comment by ilyashnil** [comment]
1. I do not know how to make unitest without specifying credentials...
3. Where I should put the documentation?

**Review by elronbandel — APPROVED** [review]

**Pull request merged**